### PR TITLE
JSON API

### PR DESCRIPTION
**Pull request by olofvi**
## PT: https://www.pivotaltracker.com/story/show/152671975
------
* generated ProductService
* generated DatastoreService
* generated ProductModel
* created a method to get all products
* created a method to show a particular product
* added basic html to show data from API
* added angular2-jsonapi

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
+  title = 'Products';
+
+  constructor(private productService: ProductService) {
+    this.getProducts();
+  }
+
+  getProducts(): void {
+    this.productService.getAll()
+      .subscribe(products => {
+        console.log(products)
```
Comment (diraulo) [inline src/app/app.component.ts]: We don't need this console.logs

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,10 +1,30 @@
...
+        this.products = products;
+      });
+  }
+  showProduct(id): void {
```
Comment (diraulo) [inline src/app/app.component.ts]: Why do we have this if it's supposed to be taken care of in a different story? 🤔  -- #FeatureCreepMaybe

**Review by diraulo — CHANGES_REQUESTED** [review]
See my comments 😉

File: src/app/app.component.spec.ts [hunk 3]
```diff
@@ -1,27 +1,47 @@
...
+//     const app = fixture.debugElement.componentInstance;
+//     expect(app.title).toEqual('app');
+//   }));
+//   it('should render title in a h1 tag', async(() => {
+//     const fixture = TestBed.createComponent(AppComponent);
+//     fixture.detectChanges();
+//     const compiled = fixture.debugElement.nativeElement;
+//     expect(compiled.querySelector('h1').textContent).toContain('Welcome to app!');
+//   }));
+// });
```
Comment (tochman) [inline src/app/app.component.spec.ts]: What's with all the comments?

**Review by tochman — COMMENTED** [review]
Just worried about the empty spec file. Perhaps we should remove it?

File: src/app/app.component.spec.ts (hunk 3 above)
Comment (olofvi) [reply src/app/app.component.spec.ts]: so the issue here is that we are not managing to make this work but Raoul is going to work on it and put that up in another PR, not sure if we should remove the file I took this code and put it in another file that is not in version control

File: src/app/service/datastore.service.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { JsonApiDatastoreConfig, JsonApiDatastore, DatastoreConfig } from 'angular2-jsonapi';
+import { Http } from '@angular/http';
+import { Product } from '../models/product';
+
+const config: DatastoreConfig = {
+  baseUrl: 'http://localhost:3000/api/v1',
+  models: {
```
Comment (tochman) [inline src/app/service/datastore.service.ts]: I would suggest making use of the `apiVersion` option. 
```
 baseUrl: 'http://localhost:3000/api',
 apiVersion: '/v1',
```

**Review by diraulo — APPROVED** [review]
Was planning on fixing the app.component.spec.ts in a separate PR, but looks like you removed it already. No biggy I might still have that in place on my local branch. 

Apart from @tochman recent comment rest looks okay to me. So if/when you address it, you can merge this PR

**Comment by olofvi** [comment]
@diraulo I have the test on my local computer in a hidden file in the project if you need the code I could send it to you :)

File: src/app/service/datastore.service.ts (hunk 4 above)
Comment (BasilMawejje) [reply src/app/service/datastore.service.ts]: Good idea. Let's fix that

**Pull request merged**